config/fetcher_test.go

```diff
@@ -0,0 +1,11 @@
+package config_test
+
+import "github.com/stretchr/testify/mock"
+
+type MockFetcher struct {
+	mock.Mock
+}
+
+func (m *MockFetcher) Get(key string) (val string) {
+	return m.Called(key).String(0)
```

Comment:
Good call. I originally wanted the mock to assert that the `Get` method was definitely being called with the correct arguments, but having a `map[string]string` buys us the same result, and lifts a dependency on the `mock` package of `testify`.

Removed the mocks in: https://github.com/github/git-lfs/pull/1419/commits/38c452b28359b64e8c4cd2600f09ba456bcf6b8f :metal: